tests/database_test.cpp

```diff
@@ -208,3 +208,216 @@ TEST_F(DatabaseTest, NoConfigTransaction) {
          ASSERT_EQ(0, tx.Read("bob").second);
        }});
 }
+
+// [Secondary Index TDD] ------------------------------------------------------
+TEST_F(DatabaseTest, CreateTable) {
+  db_ = std::make_unique<LineairDB::Database>();
+  bool success = db_->CreateTable("users");
+  ASSERT_TRUE(success);
+  bool duplicated = db_->CreateTable("users");
+  ASSERT_FALSE(duplicated);
+}
+
+//CreateSecondaryIndex("table_name", "index_name", "UNIQUE", "NOT_NULL")
+TEST_F(DatabaseTest, CreateSecondaryIndex) {
+  db_ = std::make_unique<LineairDB::Database>();
+  bool success = db_->CreateTable("users");
+  bool index_success = db_->CreateSecondaryIndex("users", "name", "UNIQUE", "NOT_NULL");
```

Comment:
Are constraints expressed in STRING?
Does this mean that the function receives variadic arguments that are all optional and unordered after the third argument?

I disagree with that design. Consider using enum classes and bitmaps.
IMO
```
db_->CreateSecondaryIndex("users", "name", LineairDB::SecondaryIndexOption::UNIQUE |  LineairDB::SecondaryIndexOption::NOT_NULL);
```